keycloak-iam/src/main/java/com/valtimo/keycloak/service/KeycloakUserManagementService.java

```diff
@@ -143,7 +145,22 @@ public List<ManageableUser> findByRole(String authority) {
 
     @Override
     public List<ManageableUser> findByRoles(SearchByUserGroupsCriteria groupsCriteria) {
-        return null;
+        Set<String> allUserGroups = new HashSet<>(groupsCriteria.getRequiredUserGroups());
+        groupsCriteria.getOrUserGroups().forEach(allUserGroups::addAll);
+
+        List<String> userIds = new ArrayList<>();
+        List<ManageableUser> allUsers = allUserGroups.stream()
+            .map(this::findByRole)
+            .flatMap(Collection::stream)
+            .filter(user -> userIds.add(user.getId()))
+            .collect(Collectors.toList());
+
+        return allUsers.stream()
+            .filter(user -> user.getRoles().containsAll(groupsCriteria.getRequiredUserGroups()))
```

Comment:
does this mean the user has to have all the configured roles for the case? i think it only has to match one. maybe check with the PO.

Reply:
nevermind. test looks good.